src/main/java/org/myrobotlab/service/TesseractOcr.java

```diff
@@ -35,70 +44,60 @@
 public class TesseractOcr extends Service {
 
   private static final long serialVersionUID = 1L;
-  transient private TessBaseAPI api;
-
+  transient private TessBaseAPI api = null;
+  // define which language model to load
+  String lang = "eng";
   public final static Logger log = LoggerFactory.getLogger(TesseractOcr.class);
 
-  public static void main(String[] args) {
-    LoggingFactory.init(Level.INFO);
-
-    try {
-
-      TesseractOcr tesseract = (TesseractOcr) Runtime.start("tesseract", "TesseractOcr");
-      // String found = tesseract.ocr("phototest.jpg");
-      // String found = tesseract.ocr("30.speed.JPG");
-      String found = tesseract.ocr("traffic.sign.jpg");
+  public TesseractOcr(String n, String id) {
+    super(n, id);
+  }
 
-      // String found = tesseract.ocr("test.jpg");
-      // String found = tesseract.ocr("test.tif");
-      log.info("found {}", found);
-      Runtime.start("gui", "SwingGui");
+  @Override
+  synchronized public void startService() {
+    super.startService();
+    initModel(lang);
+  }
 
-    } catch (Exception e) {
-      Logging.logError(e);
-    }
+  @Override
+  synchronized public void stopService() {
+    super.stopService();
+    // stop the tesseract api.
+    // TODO: additional memory/object cleanup?
+    api.End();
   }
 
-  public TesseractOcr(String n, String id) {
-    super(n, id);
+  private void initModel(String lang) {
+    api = new TessBaseAPI();
+    String tessData = "resource/TesseractOcr";
+    // load the models
+    if (api.Init(tessData, lang) != 0) {
+      log.warn("Unable to load tesseract model in {} with language {}", tessData, lang);
+      // TODO: what's the lifecycle if a service fails to start?  maybe we should throw?

```

Comment:
when the service fails to initialize it would warrant communicating back to the "user"
so the logging statement should be changed to:
warn("Unable to load tesseract model in %s with language %s", tessData, lang);  - this will log and broadcast the warning message to any UI

Reply:
done